File: TabCommon/Library/src/main/java/com/nextzy/library/base/mvvm/layer3Repository/network/operator/ErrorJsonHandling.java
package com.nextzy.library.base.mvvm.layer3Repository.network.operator;


import java.util.ArrayList;
import java.util.List;

import io.reactivex.Single;
import th.co.thekhaeng.waterlibrary.java.base.nextwork.exception.StatusCodeException;
import th.co.thekhaeng.waterlibrary.java.base.nextwork.operator.NextworkErrorJsonHandling;
import th.co.thekhaeng.waterlibrary.java.base.nextwork.reponse.NextworkResponse;
import th.co.thekhaeng.waterlibrary.java.base.structure.mvvm.layer3Repository.network.interceptor.PermissionException;
import th.co.thekhaeng.waterlibrary.java.base.structure.mvvm.layer3Repository.network.interceptor.UnAuthorizedException;


/**
 * Created by thekhaeng on 5/10/2017 AD.
 */

public class ErrorJsonHandling<T extends NextworkResponse> extends NextworkErrorJsonHandling<T>{

    public ErrorJsonHandling(Class<T> typeOfResultClass, String serviceName) {
        super(typeOfResultClass, serviceName);
    }

    @Override
    protected Single<T> checkJsonResponse(T result) {
        return Single.just(result);
    }

    @Override
    protected List<StatusCodeException> getStatusCodeException() {
        List<StatusCodeException> map = new ArrayList<>();
        map.add(new UnAuthorizedException(401));
        map.add(new PermissionException(403));
        return map;
    }
}

File: TabCommon/CommonLibrary/src/main/java/com/nextzy/library/base/mvvm/layer3Repository/network/operator/ErrorJsonHandling.kt
package com.nextzy.library.base.mvvm.layer3Repository.network.operator


import com.nextzy.library.base.mvvm.layer3Repository.network.interceptor.PermissionException
import com.nextzy.library.base.mvvm.layer3Repository.network.interceptor.UnAuthorizedException
import com.nextzy.nextwork.exception.StatusCodeException
import com.nextzy.nextwork.operator.NextworkErrorJsonHandling
import com.nextzy.nextwork.reponse.NextworkResponse
import io.reactivex.Single
import java.util.*


/**
 * Created by「 The Khaeng 」on 03 Oct 2017 :)
 */

class ErrorJsonHandling<T : NextworkResponse>(typeOfResultClass: Class<T>, serviceName: String)
    : NextworkErrorJsonHandling<T>(typeOfResultClass, serviceName) {

    override
    fun checkJsonResponse(result: T): Single<T> = Single.just(result)

    override
    fun getStatusCodeException(): List<StatusCodeException> {
        val map = ArrayList<StatusCodeException>()
        map.add(UnAuthorizedException(401))
        map.add(PermissionException(403))
        return map
    }
}
